Sources/StreamChatUI/CommonViews/InputTextView/InputTextView.swift

```diff
@@ -2,12 +2,28 @@
 // Copyright © 2021 Stream.io Inc. All rights reserved.
 //
 
+import MobileCoreServices
 import StreamChat
 import UIKit
 
+/// The delegate of the `InputTextView` that notifies when an `UIImage` is pasted.
+public protocol InputTextViewImagePasteDelegate: AnyObject {
+    /// Notifies that an `UIImage` has been pasted into the text view
+    /// - Parameters:
+    ///   - inputTextView: The `InputTextView` in which the image was pasted
+    ///   - image: The `UIImage`
+    func inputTextView(_ inputTextView: InputTextView, didPasteImage image: UIImage)
+}
+
 /// A view for inputting text with placeholder support. Since it is a subclass
 /// of `UITextView`, the `UITextViewDelegate` can be used to observe text changes.
 open class InputTextView: UITextView, AppearanceProvider {
+    /// The delegate which gets notified when an image is pasted into the text view
+    open weak var imagePasteDelegate: InputTextViewImagePasteDelegate?
```

Comment:
hm... how about
```suggestion
    open var inputViewViewDelegate: InputTextViewImagePasteDelegate? {
     delegate as? InputTextViewDelegate
    }
```

Reply:
I think merging the delegates, will be confusing IMO. Because we will still have the `inputTextView.delegate` and then the `inputTextView.inputTextViewDelegate` too. 

Maybe can make the other delegate a bit more generic like `clipboardAttachmentDelegate` something like this.. 🤔 Where we put all possible attachments from the clipboard in one delegate